check_license_go_code.sh

```diff
@@ -16,16 +14,29 @@ cat > license.tmp <<EOF
 //    limitations under the License.
 EOF
 lines=$(cat license.tmp | wc -l)
+# we need to add two extra lines missing from the license preamble
+# // Copyright <copyrigt_year> Northern.tech AS
+# //
+lines=$(($lines + 2))
 
 ret=0
 for each in $(find . -type f \( ! -regex '.*/\..*' ! -path "./Godeps/*" ! -path "./vendor/*" -name '*.go' \)); do
-  echo "Checking $each for correct license header"
-  head -n $lines $each | diff -qu license.tmp - > /dev/null
+  modified_year=$(date +%Y -r $each)
+  
+  echo "Checking $each for correct license header; last modified in $modified_year"
+
+  head -n $lines $each | tail -n +3 | diff -qu license.tmp - > /dev/null
   if [ ! "$?" -eq "0" ]; then
     echo "!!! FAILED license check on $each"
     ret=1
   else
-    echo "License check passed on $each"
+    copyright_modified=$(echo "// Copyright <copyright_year> Northern.tech AS" | sed "s/<copyright_year>/$modified_year/g")
+    copyright_file=$(head -n 1 $each)
+    if [ "$copyright_modified" == "$copyright_file" ]; then
+      echo "License check passed on $each"
+    else
+      echo "!!! FAILED license check on $each; copytight year does not match modification year"
```

Comment:
Propose switching `copytight year does not match modification year` with `make sure copyright year matches last modified year of the file`.

Reply:
It's your baby, so will do as you want :-)